doctoshotgun.py

```diff
@@ -94,6 +95,34 @@ def iter_centers_ids(self):
             data = json.loads(div.attrib['data-props'])
             yield data['searchResultId']
 
+    def get_next_page(self):
+        # French doctolib uses data-u attribute of span-element to create the link when user hovers span
+        for span in self.doc.xpath('//div[contains(@class, "next")]/span'):
+            if not span.attrib.has_key('data-u'):
+                continue
+
+            # How to find the corresponding javascript-code:
+            # Press F12 to open dev-tools, select elements-tab, find div.next, right click on element and enable break on substructure change
+            # Hover "Next" element and follow callstack upwards
+            # JavaScript:
+            # var t = (e = r()(e)).data("u")
+            #     , n = atob(t.replace(/\s/g, '').split('').reverse().join(''));
+            
+            import base64
+            href = base64.urlsafe_b64decode(''.join(span.attrib['data-u'].split())[::-1]).decode()
```

Comment:
Wow, that is crazily complex. I tried to understand why this is necessary because if I just check the [source code of the page](view-source:https://www.doctolib.fr/vaccination-covid-19/paris?ref_visit_motive_ids[]=6970&ref_visit_motive_ids[]=7005&ref_visit_motive_ids[]=7107&ref_visit_motive_ids[]=7945&ref_visit_motive_ids[]=8741), I directly see the link as a regular `a` tag of the _Suivant_ button.

```html
<div class="next dl-rounded-borders dl-white-bg"><a href="/vaccination-covid-19/paris?page=2&amp;ref_visit_motive_ids%5B%5D=6970&amp;ref_visit_motive_ids%5B%5D=7005&amp;ref_visit_motive_ids%5B%5D=7107&amp;ref_visit_motive_ids%5B%5D=7945&amp;ref_visit_motive_ids%5B%5D=8741">Suivant

Maybe you can give a brief explanation.
```

Reply:
My tests are based on this [site](https://www.doctolib.fr/vaccination-covid-19/france?ref_visit_motive_ids[]=6970&ref_visit_motive_ids[]=7005).
It seems that the next-link in some french groups are encoded and some are just plain `a` tags, link in your link.
Maybe i have trigged some bot-detection, as more and more groups getting _encoded_ as `span`

Steps to reproduce:

1. Reload the page, do **not** hover the _Suivant_ button
2. find button in dev-tools:
![image](https://user-images.githubusercontent.com/2452341/124794473-5e27a900-df4f-11eb-9b78-fa1e932945c9.png)
3. Now you can hover the _Suivant_ button and the some magic happen
![image](https://user-images.githubusercontent.com/2452341/124795827-c925af80-df50-11eb-91e3-82822821eb7d.png)

https://user-images.githubusercontent.com/2452341/124795564-75b36180-df50-11eb-95b8-49343a43a8f7.mp4

`base64.urlsafe_b64decode(''.join(span.attrib['data-u'].split())[::-1]).decode()` is the python version of the following javascript code:
![image](https://user-images.githubusercontent.com/2452341/124797979-2ae71900-df53-11eb-9692-131b961a5997.png)

Reply:
Understood, thanks a lot for the explanation. This website is really doing some weird stuff.